macros/first_value.sql

```diff
@@ -0,0 +1,19 @@
+{% macro first_value(first_value_field, partition_field, order_by_field, order="asc") -%}
+
+{{ adapter.dispatch('first_value', packages = fivetran_utils._get_utils_namespaces()) (first_value_field, partition_field, order_by_field, order="asc") }}
+
+{%- endmacro %}
+
+--Default first_value calculation
+{% macro default__first_value(first_value_field, partition_field, order_by_field, order="asc")  %}
+
+    first_value( {{ first_value_field }} ) over (partition by {{ partition_field }} order by {{ order_by_field }} {{ order }} )
+
+{% endmacro %}
+
+--first_value calculation specific to Redshift
+{% macro redshift__first_value(first_value_field, partition_field, order_by_field, order="asc") %}
+
+    first_value( {{ first_value_field }} ) over (partition by {{ partition_field }} order by {{ order_by_field }} {{ order }} , {{ partition_field }} rows unbounded preceding )
```

Comment:
is ordering by the partition_field necessary in redshift?

Reply:
Strangely yes. When I remove the partition by statement I receive the first value for for the first conversation (in Intercom terms) instead of the first value for each individual conversation. I agree it does not seem necessary to include, but does not provide reliable results when removed.